### PR TITLE
Do retry HTTP 400 only in case of throttling errors.

### PR DESCRIPTION
**Pull request by kkuzmin**
Look inside 400 error message to figure out if it is a throttling error in order to decide weather to retry the request or not.

### CODE REVIEW

File: src/erlcloud_aws.erl [hunk 1]
```diff
@@ -511,3 +518,17 @@ get_filtered_statuses(ServiceNames, Statuses) ->
...
     Statuses).
+
+-spec is_throttling_error_response(aws_request()) -> true | false.
+is_throttling_error_response(RequestResponse) ->
+    #aws_request{
+         response_type = error,
+         error_type = aws,
+         response_body = RespBody} = RequestResponse,
+  
+    case re:run(RespBody, <<"Throttling">>) of
```
Comment (kkuzmin) [inline src/erlcloud_aws.erl]: Any ideas if we need more complicated error message parsing here are welcome.

Comment (motobob) [inline src/erlcloud_aws.erl]: i have no better ideas given the amount of case matching we need to do to cover them all

Comment (danilagamma) [inline src/erlcloud_aws.erl]: `binary:match/2` is better than regex here.

Comment (motobob) [inline src/erlcloud_aws.erl]: agree

File: src/erlcloud_s3.erl [hunk 2]
```diff
@@ -1200,11 +1200,17 @@ s3_result_fun(#aws_request{response_type = ok} = Request) ->
...
+s3_result_fun(#aws_request{response_type = error,
+                           error_type = aws,
+                           response_status = Status} = Request) when
 %% Retry for 400, Bad Request is needed due to Amazon 
 %% returns it in case of throttling.
-%% Also retry conflictin operations 409,Conflict.
-      Status =:= 400; Status =:= 409; Status >= 500 ->
-    Request#aws_request{should_retry = true};
+      Status =:= 400 ->
+    ShouldRetry = erlcloud_aws:is_throttling_error_response(Request),
```
Comment (danilagamma) [inline src/erlcloud_s3.erl]: Is `s3` API throttled at all? I don't see anything mentioning it here: http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html

Comment (motobob) [inline src/erlcloud_s3.erl]: funny enough it's does not but it has :bomb: 

```
SlowDown    Reduce your request rate.   503 Slow Down   Server
```

Comment (kkuzmin) [inline src/erlcloud_s3.erl]: I'll remove 400 case then as we already have retries for 503.

**Pull request merged**